lua/orgmode/org/mappings.lua

```diff
@@ -462,6 +462,111 @@ function OrgMappings:show_help()
   return Help.show()
 end
 
+function OrgMappings:edit_special()
+  local start_bufnr = vim.api.nvim_get_current_buf()
+  local start_pos = vim.api.nvim_win_get_cursor(0)
+  local file = Files.get_current_file()
+
+  local make_temp_buf = function()
+    local option = config.org_src_window_setup
+    if type(option) == 'string' then
+      vim.cmd(option)
+    elseif type(option) == 'function' then
+      option()
+    else
+      utils.echo_error("Invalid 'org_src_window_setup' setting, value is not a 'string' or 'function'")
+      return false
+    end
+
+    return vim.api.nvim_get_current_buf()
+  end
+
+  -- The block might not have content yet, which is fine
+  local block = utils.get_nearest_src_block(file, start_pos)
+  if not block then
+    return
+  end
+
+  -- Assume that the content of the source block will only change
+  -- below the first line of the block
+  local src_start_line = block.name.node:range()
+  local content = block.contents and block.contents.text_list or {}
+  local ft = block.parameters.text
+
+  local bufnr = make_temp_buf()
+  if not bufnr or not vim.api.nvim_buf_is_valid(bufnr) then
+    utils.echo_info('Cancel editing source block, invalid buffer')
+    return
+  end
+
+  vim.cmd('e ' .. vim.fn.tempname())
+
+  local ctx = {
+    bufnr = bufnr,
+    filename = file.filename,
+    src_start_line = src_start_line + 1,
+    start_bufnr = start_bufnr,
+    start_pos = start_pos,
+  }
+
+  vim.api.nvim_buf_set_option(bufnr, 'bufhidden', 'wipe')
+  vim.api.nvim_buf_set_option(bufnr, 'filetype', ft)
+  vim.api.nvim_buf_set_lines(bufnr, 0, -1, false, content)
+  vim.api.nvim_buf_set_option(ctx.bufnr, 'modified', false)
+  vim.api.nvim_buf_set_var(bufnr, '__org_edit_special_ctx', ctx)
+
+  vim.cmd(
+    string.format(
+      [[autocmd BufWipeout <buffer=%s> ++once lua require('orgmode').action('org_mappings._edit_special_callback')]],
+      bufnr
+    )
+  )
+end
+
+function OrgMappings:_edit_special_callback()
+  local success, ctx = pcall(vim.api.nvim_buf_get_var, 0, '__org_edit_special_ctx')
+  if not success or ctx.bufnr ~= vim.api.nvim_get_current_buf() then
+    return { message = '' }
```

Comment:
Need to double check if the return value here is supposed to actually specify this, as I believe it is just a part of the `pcall` result when there is an error.